pandas/core/missing.py

```diff
@@ -128,6 +128,43 @@ def clean_interp_method(method, **kwargs):
     return method
 
 
+def find_valid_index(values, how: str):
+    """
+    Retrieves the index of the first valid value.
+
+    Parameters
+    ----------
+    values : ndarray or ExtensionArray
+    how : {'first', 'last'}
+        Use this parameter to change between the first or last valid index.
+
+    Returns
+    -------
+    int or None
+    """
+    assert how in ["first", "last"]
```

Comment:
I realize this is the existing behavior, but does it make sense to change this to a `ValueError` instead?  That seems more consistent with how we've been validating these types of parameters elsewhere.